.github/workflows/publish-crate.yml

```diff
@@ -0,0 +1,15 @@
+name: publish-crate
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  crate:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: cargo login ${LE_AUTOMATON_CRATES_IO_API_TOKEN}
+      env:
+        CRATES_IO_TOKEN: ${{ secrets.LE_AUTOMATON_CRATES_IO_API_TOKEN }}
```

Comment:
Why do these use two different styles of substitution? I'm guessing the `${ }` one is just shell expansion and that env var is automatically set by Github, but the second with `${{ secrets.FOO }}` is a yaml-level template substitution?

Reply:
Yeah, the `secrets.` one fetches the token from the secrets configured on the (if you have the right privs on this repo you can [take a look](https://github.com/abetterinternet/libprio-rs/settings/secrets)) and injects its value into the environment of the `run` action. However I think I have this wrong because the environment variable will be `CRATES_IO_TOKEN`. But I need to test this before merging.

Reply:
Ah, I had assumed that all variables were added to the environment by default, but you were also aliasing it to `CRATES_IO_TOKEN` because it was also required by something else.

Looks like you don't actually need to call `cargo login` though: that just sets the token in the persistent configuration. For CI we can either pass `--token` to `publish`, or just set [`CARGO_REGISTRY_TOKEN`](https://doc.rust-lang.org/cargo/reference/config.html#registrytoken) and that's equivalent to calling `login` first (but non-persistent).

Reply:
Ah, I had assumed that all variables were added to the environment by default, but you were also aliasing it to `CRATES_IO_TOKEN` because it was also required by something else.

Looks like you don't actually need to call `cargo login` though: that just sets the token in the persistent configuration. For CI we can either pass `--token` to `publish`, or just set [`CARGO_REGISTRY_TOKEN`](https://doc.rust-lang.org/cargo/reference/config.html#registrytoken) and that's equivalent to calling `login` first (but non-persistent).